### ISSUE TITLE
[v0.9] pimatic sometimes exits on restart via UI

### ISSUE THREAD

**Issue by mwittig** [pimatic/pimatic#862]
If pimatic is restarted via UI after a plugin has been activated it sometimes exits. There is no error message in the log, last message seen:
```
11:38:56.374 [pimatic] info: Flushing database to disk, please wait...
11:38:56.438 [pimatic] info: Flushing database to disk, please wait... Done.
```

**Comment by sweebee** [comment]
+1 same here. If i click restart on the plugin page.

**Comment by sweebee** [comment]
Today I had to restart it manually again but that failed:

```
root@pimatic-thuis:/pimatic-app# service pimatic restart
Job for pimatic.service failed. See 'systemctl status pimatic.service' and 'journalctl -xn' for details.
```

**Comment by sweebee** [comment]
Lets bumb this one. Stil a very big issue?! Happens to me like 90% of the time.

**Comment by mwittig** [comment]
Sweetpi wasn't able to reproduce it. In my setup it happens frequently. Question: Do you have a "development" or "production" setup for pimatic? Mine is a "development" and maybe this is causing to happen the issue frequently.

**Comment by abmantis** [comment]
Happens here every time I restart it trough the web interface. Using development setup I think.

**Comment by sweetpi** [comment]
I finally found the problem and fixed it in v0.9.28.
The problem was the following:
For restarting we called "path-to-pimatic/pimatic start" and then terminated the process. So there is a race condition: If the process terminated before the start command checks the pid file pimatic starts successfully. However if the start command checks the pid file before the process terminates, then it will not start pimatic, because it is already running (until it terminates...). 

The fix: don't terminate the process and only call "path-to-pimatic/pimatic restart". The restart command terminates the running instance and start the new one :)
